projects/igniteui-angular/src/lib/tabs/tab-item.component.ts

```diff
@@ -168,41 +172,78 @@ export class IgxTabItemComponent extends IgxTabItemBase implements AfterViewInit
      */
     @HostListener('click', ['$event'])
     public onClick(event) {
-        this.select();
+        if (this.autoGenerated) {
+            this.select();
+        }
     }
 
     /**
      * @hidden
      */
-    @HostListener('keydown.arrowright', ['$event'])
-    public onKeydownArrowRight(event: KeyboardEvent) {
-        this.onKeyDown(false);
-    }
+    @HostListener('keydown', ['$event'])
+    public onKeydown(event: KeyboardEvent) {
+        event.preventDefault();
```

Comment:
We should not call preventDefault() here since this is stopping the tab navigation.